WalletWasabi.Gui/Behaviors/CloseWindowBehavior.cs

```diff
@@ -0,0 +1,45 @@
+using Avalonia;
+using Avalonia.Controls;
+using Avalonia.Xaml.Interactivity;
+using ReactiveUI;
+using System;
+using System.Reactive.Disposables;
+using System.Reactive.Linq;
+
+namespace WalletWasabi.Gui.Behaviors
+{
+	public class CloseWindowBehavior : Behavior<Window>
+	{
+		public static readonly StyledProperty<bool> CloseTriggerProperty =
```

Comment:
This is the wrong pattern...

Properties are not for triggering actions... the correct way to do this would be something like:

1) add an `ICloseable` interface on the Window with a single Close method.
2) On the viewmodel add a command CloseWindowCommand with a parameter type ICloseable.
3) on the view add Bind to the command and use the Window itself (via x:Name) as the command parameter.
4) remove this behavior class.

Reply:
Actually we can do this without an interface on the view.

1) add a behaviour with a command. CloseWindowCommandBehavior

2) in vm create the command but it does nothing in the handler.

3) in the behaviour on attached when the command changes, subscribe to the command and close when the command executes.


I’d say either of those options is good, whichever you prefer.

Reply:
Done here: https://github.com/zkSNACKs/WalletWasabi/pull/3733/commits/b4e2480ec1fd9848f220bf2cf0567606881a3e16